### ISSUE TITLE
BaseException: Could not construct partition: vector::_M_default_append 

### ISSUE THREAD

**Issue by houruiyan** [atarashansky/SAMap#93]
Dear Dr, thank you very much for this useful package. I just use the raw data to input samap. However it has this error.
<img width="1133" alt="image" src="https://user-images.githubusercontent.com/56068158/206887473-ebed6005-fcd6-461c-84c8-926e66bf312b.png">
My data looks like this
<img width="795" alt="image" src="https://user-images.githubusercontent.com/56068158/206887531-d3b45d31-b38e-4d47-94f5-8f616042cbbd.png">

This adata comes from different replicates. I just use concatenate from anndata to merge them. 

There is my code. Hope to get your help. Thank you very much!

````
import pandas as pd
import scanpy as sc
from samap.mapping import SAMAP
from samalg import SAM
from samap.analysis import (get_mapping_scores, GenePairFinder, transfer_annotations,
                            sankey_plot, chord_plot, CellTypeTriangles, 
                            ParalogSubstitutions, FunctionalEnrichment,
                            convert_eggnog_to_homologs, GeneTriangles)

mouse = '/storage/yhhuang/users/ruiyan/VAE/four_species/finalAdata/mouse_adata.h5ad'
rat = '/storage/yhhuang/users/ruiyan/VAE/four_species/finalAdata/rat_adata.h5ad'
pig = '/storage/yhhuang/users/ruiyan/VAE/four_species/finalAdata/pig_adata.h5ad'
human='/storage/yhhuang/users/ruiyan/VAE/four_species/finalAdata/human_adata.h5ad'


adata=sc.read('/storage/yhhuang/users/ruiyan/VAE/four_species/finalAdata/mouse_adata.h5ad')
adata

filenames = {'mouse':mouse,'rat':rat,'pig':pig,'human':human}

sm = SAMAP(
        filenames,
        f_maps = '/storage/yhhuang/users/ruiyan/VAE/four_species/SAMap/map/maps',
        save_processed=True
    )`
```

**Comment by atarashansky** [comment]
Can you try the following?

```
from samalg import SAM
sam=SAM(counts=adata) #adata is the raw mouse data
sam.preprocess_data()
sam.run()
sam.leiden_clustering()
```

Does the above work?

**Comment by houruiyan** [comment]
Hi Alexander, 

thank you very much for your answer! Now I think it is ok. I found out the reason to cause this problem. I think it because the version of python-igraph is not correct! 
Because when I run this command `sc.tl.leiden(adata,use_weights=False)` , I also got the error. Finally, when I fix this question and change the version of python-igraph, the Samap can run smoothly.

https://github.com/scverse/scanpy/issues/2339#issuecomment-1261132252
